docs/backends/lua2.rst

```diff
@@ -0,0 +1,173 @@
+Lua Backend
+===========
+
+* Native: Yes
+* Master: Yes
+* Slave: No
+* Superslave: No
+* Autoserial: No
+* DNSSEC: Yes
+* Disabled data: No
+* Comments: No
+* Module name: lua2
+* Launch name: ``lua2``
+
+This is a rewrite of existing Lua backend.
+This backend is stub between your Lua script and PowerDNS authoritative server.
+The backend uses AuthLua4 base class, and you can use same functions and types as in any other Lua script.
+
+.. warning::
+  Some of the function calls and configuration settings have been changed, please review this document carefully.
+
+.. warning::
+  All places which use DNS names now use DNSName class which cannot be compared directly to a string.
+  To compare them against a string use either ``tostring(dnsname)`` or ``newDN(string)``.
+
+API description (v1)
+^^^^^^^^^^^^^^^^^^^^
+
+``bool dns_dnssec``
+~~~~~~~~~~~~~~~~~~~
+If your script supports DNSSEC, set this to true.
+
+``dns_lookup(qtype, qname, domain_id)``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Perform lookup of given resource record name and type.
+
+INPUT:
+ - string qtype - Type of queried resource record
+ - DNSName qname - Name of queried resource record
+ - int domain_id - ID of associated domain
+
+OUTPUT:
+ Expects a table which has rows of following tables:
+ - DNSName name - resource record name (can also be string)
+ - string type - type of resource record
+ - string content - resource record content
+ - int ttl - time to live for this resource record (default: configured value)
+ - int domain_id - ID of associated domain (default: -1)
+ - bool auth - Whether data is authoritative or not (default: true)
+ - int last_modified - UNIX timestamp of last modification
+ - int scope_mask - How many bytes of source IP netmask was used for this result
```

Comment:
What use is this if `lookup` does not get passed the source IP?

Reply:
I guess I could pass the DNSPacket here... or maybe a special table with values of interest?

Reply:
Table makes more sense to me.

Reply:
special table added.